aws/resource_aws_ecs_service.go

```diff
@@ -857,9 +862,24 @@ func expandPlacementStrategy(s []interface{}) ([]*ecs.PlacementStrategy, error)
 	}
 	pss := make([]*ecs.PlacementStrategy, 0)
 	for _, raw := range s {
-		p := raw.(map[string]interface{})
-		t := p["type"].(string)
-		f := p["field"].(string)
+		p, ok := raw.(map[string]interface{})
```

Comment:
interesting, I didn't know that you could get a slice with nil map`[nil...]` from a config where a required field was sent, even if as an empty string.

Reply:
😅  Yeah, its a surprising feature of the Terraform Plugin SDK. If all nested attributes match the zero value for their type, then the map itself is `nil` and type assertion panics.